packages/core/src/internal-components/panel/panel.element.ts

```diff
@@ -0,0 +1,48 @@
+import { LitElement, html } from 'lit-element';
+import { baseStyles } from '@cds/core/internal';
+import { styles } from './panel.element.css.js';
+
+export const CdsInternalPanelTagName = 'cds-internal-panel';
+
+/**
+ *
+ * ```typescript
+ * import '@cds/core/internal-components/panel/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-internal-panel>
+ *  // content
+ * </cds-internal-panel>
+ * ```
+ *
+ * @beta
+ * @element cds-internal-panel
+ * @slot - Content slot for the content inside the panel
+ * @cssprop --width
+ * @cssprop --height
+ * @cssprop --color
+ * @cssprop --overflow
+ * @cssprop --overflow-x
+ * @cssprop --overflow-y
+ * @cssprop --background
+ * @cssprop --border
+ * @cssprop --border-radius
+ * @cssprop --padding
+ * @cssprop --box-shadow
+ */
+export class CdsInternalPanel extends LitElement {
+  static get styles() {
+    return [baseStyles, styles];
+  }
+
+  render() {
+    const layout = this.getAttribute('cds-layout') || 'vertical';
+
+    return html`
+      <div class="private-host" cds-layout="${layout}">
```

Comment:
Since this is across the shadow boundary, I'm curious as to how this translates. Thoughts, @coryrylan ?

Reply:
Yeah this will be an issue. The layout util cannot be dynamically assigned like this since the unused CSS is compiled/treeshaken out of the templates. Also the CSS can only apply to one level, it cant cross boundaries easily between shadow/light DOM. It would need to be something like this https://github.com/vmware/clarity/blob/next/packages/core/src/modal/modal-actions.element.ts#L40

Reply:
Seems like we are getting back to the initial implementation of `render` method? Also if we do what we did in the `modal-action` element we will get back to the previous situation where the component won't have a `private-host` class.

Reply:
`.private-host` in this component can still be useful, even if it just a wrapper for a `slot`.

In theory, people could style the background, border, border-radius and padding of the panel through the CSS custom properties. All of those would/should be applied to `.private-host`.

Reply:
If i understand you correctly we will get back to the concern that was mentioned [here](https://github.com/vmware/clarity/pull/5538#discussion_r580437399). 

When I verified the concerns is on point I updated the implementation to resolve the issue we were about to face and I end up with the current implementation. 

Example of the issues with the previous implementation 

```
 <cds-panel cds-layout="horizontal">
    <div class="private-host> // <- the only element `cds-layout` will apply styles to
       //content 
    </div>
 </cds-panel>
```

I might be missing something but this is how I see it currently.

Reply:
No, that's not what I'm getting at. What I'm saying is...

1) have the `<cds-panel>` greedily take up any space (height/width) available to it
2) have _no cds-layout_ specified on the `cds-panel` component
3) have a `div.private-host` internal to the `cds-panel`. Why? so that people could potentially use CSS or cds-layout padding utilities to force the `.private-host` to have, basically, a margin of its own.
4) End users would then use `cds-layout` on the contents that are then rendered in the slot inside `.private-host`. So if they want it to be vertical/horizontal, cool. Go for it.

As for how this translates to cards, you'd likely want to do either what we are doing in `form-group.element.ts` with the `layout` property. But I don't think it's needed for a panel. Panels are more generic. Cards are more specific and a little more complicated. I could see forcing a layout on panels via a property as complicating the development of things like drawers on the vertical axes or even the detail pane of a datagrid (both of which should/would use the panel as a base).

Reply:
Got it!